### PR TITLE
Modernized infrastructure and dependencies. 

### PR DESCRIPTION
**Pull request by arunkpatra**
### Updated the following

- Java: The codebase would now build just fine against JDK 11. Gradle plugins have been updated.
- Jacoco Plugin
- Spring Boot

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -29,6 +29,7 @@ subprojects {
     targetCompatibility = JavaVersion.VERSION_1_8
     checkstyle {
         toolVersion = CheckStyleVersion
+        configDir = file("$rootProject.projectDir/quality-checks/checkstyle")
```
Comment (arunkpatra) [inline build.gradle:32]: The checkstyle plugin needs this config to build against JDK11 due to tighter security restrictions imposed by Java 11.

File: build.gradle [hunk 2]
```diff
@@ -72,9 +73,9 @@ task alljavadoc(type: Javadoc) {
 
 task jacocoRootReport(type: JacocoReport) {
     dependsOn = subprojects.test
-    sourceDirectories = files(subprojects.sourceSets.main.allSource.srcDirs)
-    classDirectories = files(subprojects.sourceSets.main.output)
-    executionData = files(subprojects.jacocoTestReport.executionData)
+    sourceDirectories.from = files(subprojects.sourceSets.main.allSource.srcDirs)
```
Comment (arunkpatra) [inline build.gradle:76]: These changes are needed due to Gradle and Jacoco version updates

File: quality-checks/checkstyle/google_checks.xml [hunk 3]
```diff
@@ -407,7 +409,8 @@ page at http://checkstyle.sourceforge.net/config.html -->
   </module>
 
   <module name="SuppressionFilter">
-      <property name="file" value="./quality-checks/checkstyle/checkstyle-suppressions.xml"/>
```
Comment (arunkpatra) [inline quality-checks/checkstyle/google_checks.xml:410]: This is needed to satisfy JDK 11 requirements.

File: sample-rest-service/build.gradle [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
-apply plugin: 'java'
+apply plugin: 'java-library'
 apply plugin: 'org.springframework.boot'
 
 dependencies {
-    compile project(':algorithms')
-    compile('org.springframework.boot:spring-boot-starter-web')
-    testCompile('org.springframework.boot:spring-boot-starter')
-    testCompile('io.rest-assured:spring-mock-mvc:' + RESTAssuredVersion)
+    api platform("org.springframework.boot:spring-boot-dependencies:${SpringBootGradlePluginVersion}")
```
Comment (arunkpatra) [inline sample-rest-service/build.gradle:7]: Newer style. Old style deprecated.

File: sample-rest-service/src/main/java/com/sl/sample/rest/service/SampleRESTApplication.java [hunk 5]
```diff
@@ -2,12 +2,10 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
-import org.springframework.context.annotation.ComponentScan;
 import org.springframework.web.servlet.config.annotation.EnableWebMvc;
 
 @SpringBootApplication
 @EnableWebMvc
-@ComponentScan
```
Comment (arunkpatra) [inline sample-rest-service/src/main/java/com/sl/sample/rest/service/SampleRESTApplication.java:10]: Not needed anymore; underlying Spring Boot enghancements make this annotation redundant.

Comment (sachinlala) [reply sample-rest-service/src/main/java/com/sl/sample/rest/service/SampleRESTApplication.java:10]: yes good/nice

**Pull request merged**